core/src/main/java/org/apache/iceberg/OverwriteData.java

```diff
@@ -56,14 +67,34 @@ public OverwriteFiles addFile(DataFile file) {
   }
 
   @Override
-  public OverwriteFiles validateAddedFiles() {
-    this.validateAddedFiles = true;
+  public OverwriteFiles deleteFile(DataFile file) {
+    Preconditions.checkState(!overwriteByRowFilter,
+        "Either overwriteByRowFilter or deleteFile should be used, not both");
+    this.deleteFilesByPath = true;
+    delete(file.path());
+    return this;
+  }
+
+  @Override
+  public OverwriteFiles validateAddedFilesMatchOverwriteFilter() {
+    Preconditions.checkState(!deleteFilesByPath,
+        "Added files can be validated only if overwriteByRowFilter is used");
```

Comment:
I don't think these are mutually exclusive. I could still delete by filter and want to validate no files matching a filter (probably the same one) have been added.